app/javascript/components/workers-form/workers-form.jsx

```diff
@@ -20,57 +20,55 @@ const WorkersForm = ({ server: { id, name }, product, zone }) => {
       .then(({ workers: { worker_base } }) => {
         const wb = worker_base;
 
-        const countDefault = wb.defaults.count;
-        const memDefault = toBytes(wb.defaults.memory_threshold);
-        const baseMemDefault = toBytes(wb.queue_worker_base.defaults.memory_threshold);
-        const monitorDefault = toBytes(wb.event_catcher.defaults.memory_threshold);
-
-        const selectCount = value => (typeof value === 'number' ? value : countDefault);
+        const baseCount = wb.defaults.count;
+        const queueCount = wb.queue_worker_base.defaults.count || baseCount;
+        const baseMem = toBytes(wb.defaults.memory_threshold);
+        const queueMem = toBytes(wb.queue_worker_base.defaults.memory_threshold) || baseMem;
 
         const parsedValues = {
           generic_worker: {
-            memory_threshold: parseWorker(wb.queue_worker_base.generic_worker).bytes || baseMemDefault,
-            count: selectCount(wb.queue_worker_base.generic_worker.count),
```

Comment:
this UI code knows way too much about the worker class hierarchy.  In other words, if we ever change it, we need to be mindful of this area of code.  Also, wondering if changing same via our API is possible and, if so, how safe that code is.

Reply:
Yeah, this is exactly the problem here.  We're reimplementing the backend logic in the frontend.

Apparently, we don't have an api for getting worker settings yet.  Unfortunately, the current code is really bad, it's the source of a user accidentally changing lots of workers to use 200 MB memory threshold, so we need to fix this.  I'm not sure when this was broken.

Reply:
💯 agree - unfortunately, we don't have an API that normalizes the worker settings, and that would be the "real" fix

Reply:
yeah, thanks... I should have clarified, the UI is calling `API.get(`/api/servers/${id}/settings`)`, so we have the API for getting server settings but this does not walk the workers subhash to resolve each subclass's settings based on inherited values.

This isn't the first time that this special inheritance structure of the workers subhash has caused confusion and complicated code.  Even if we flatten worker settings, we still have this inheritance that isn't always obvious for most users.  

Is it time to consider if it makes sense doing inheritance at all in the workers hash?  I believe the original idea was to enable setting more general defaults that get inherited but if this is not easy, is it buying anything for us other than creating complicated code?

Reply:
I spoke with LJ and I'm leaning towards removing the inheritance as well.  If we did so, we'd need to update this UI to present all of the workers and allow them to edit it.  In its current presentation it is not feasible but perhaps in a tabular form it would make more sense.

Technically we'd lose "bulk" changing of all workers of a particular class, but it's arguable if that is useful.  I'd think users would only change values where there are having problems, not necessarily across the board.  Even so, it's not hard to go in and change, at most, 20 workers to the same value. @chessbyte Thoughts?

Reply:
cc @agrare Would like your perspective too, because this overlaps with providers a lot

Reply:
No issues on my side re: removing the inheritance.  The only thing I can see being more complex (and this is just on the developer side so we can manage it) is if we want to add a setting that is used on e.g. the base MiqWorker class we have to blast it out to every worker config section instead of just adding in one place.

Reply:
I'd rather take on that developer inconvenience that push it onto users.  In fact, that makes me lean even _more_ to removing the inheritance.

Reply:
Yeah and something we can do a shared provider spec test for, e.g. make sure all provider queue workers have `:dequeue_method`, `:poll_method`, and `:queue_timeout` settings